models/forms/TaskCreateForm.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace app\models\forms;
+
+use Yii;
+use TaskForce\exceptions\DataSaveException;
+use yii\base\Model;
+use app\models\Category;
+use app\models\File;
+use app\models\Task;
+
+class TaskCreateForm extends Model
+{
+    public $taskName;
+    public $taskEssence;
+    public $taskDetails;
+    public $category;
+    // public $location;
+    public $taskBudget;
+    public $taskDeadline;
+    public $files;
+
+    public function attributeLabels()
+    {
+        return [
+            'taskName' => 'Название задания',
+            'taskEssence' => 'Мне нужно',
+            'taskDetails' => 'Подробности задания',
+            'category' => 'Категория задания',
+            //'location' => 'Локация',
+            'taskBudget' => 'Бюджет',
+            'taskDeadline' => 'Срок исполнения',
+            'files' => 'Файлы',
+        ];
+    }
+
+    public function rules()
+    {
+        return [
+            [['taskName', 'taskEssence', 'taskDetails', 'category'], 'required'],
+            ['taskName', 'string', 'min' => 3, 'max' => 50],
+            ['taskEssence', 'string', 'min' => 10, 'max' => 255],
+            ['taskDetails', 'string', 'min' => 10],
+            [['taskDeadline'], 'date', 'when' => function ($form) {
+                return strtotime($form->taskDeadline) < time();
+            }, 'message' => 'Дата рождения должна быть не меньше текущей даты'],
+            ['category', 'exist', 'targetClass' => Category::class, 'targetAttribute' => ['category' => 'category_id']],
+            ['taskBudget', 'integer', 'min' => 0],
+            [['files'], 'file', 'skipOnEmpty' => true, 'maxFiles' => 10],
+        ];
+    }
+
+    public function addTask()
+    {
+        $task = new Task();
+
+        $task->task_status = Task::STATUS_NEW;
+        $task->task_name = $this->taskName;
+        $task->task_essence = $this->taskEssence;
+        $task->task_details = $this->taskDetails;
+        $task->category_id = $this->category;
+        $task->task_deadline = $this->taskDeadline;
+        $task->task_budget = $this->taskBudget;
+        $task->customer_id = Yii::$app->user->id;
+        $task->save();
+
+        return  $task;
+    }
+
+    public function uploadFiles($taskId)
+    {
+        if ($this->validate() && !empty($this->files)) {
+            foreach ($this->files as $file) {
+                // Базовое имя будет использоваться для публичного отображения имени файла в удобочитаемом формате на усмотрение пользователя
+                $addedFileBaseName = $file->baseName;
+                // Уникальное именя файла в БД 
+                $addedFileName = time() . $addedFileBaseName . '.' . $file->extension;
```

Comment:
Исходное имя файла (`$addedFileBaseName`) при загрузках лучше не использовать. `time()` возвращает секунды. Что будет, если железо окажется настолько мощным, что за 1 секунду будут загружаться несколько файлов сразу? Одинаковые имена, перезапись одного файла другим.

Можно так:

```suggestion
                $addedFileName = md5(microtime(true)) . '.' . $file->extension;
```